### ISSUE TITLE
Editor Performance improvement: Store Thumbnails of every sprite imported into the game

### ISSUE THREAD

**Issue by selmiak** [adventuregamestudio/ags#1429]
**Describe the problem**
I use sprites as big as full game screen, that might be 640x400 or even bigger and I am probably not the only one to use full screen animations. when scrolling through the sprites or the views it lags. if AGS had generated thumbnails of all these images on import that would run way smoother.

**Suggested change**
have a setting in the game settings if you want AGS to create thumbnails for every imported sprite or not, maybe even ask when creating a new game project and ask for the filepath and then do it. It is a question of convenience over disc space, and I know what I would choose every time. Then, when importing the sprites into the game also generate a thumbnail that gets shown in the sprites view and in the views and we can scroll smooth again as AGS does not have to import sequences of full game size images on the fly, scale them down and draw that but can just show fitting images.
the size of the thumbnails could also be set in the game settings as the current thumbnails are really small.

**Comment by ivan-mogilko** [comment]
This sounds like a serious change to the Sprite viewer, and because currently editor reuses engine's "sprite cache" (from the "native code" library), that would also require writing editor's own sprite storage in C#. Which in turn means it's best to postpone this until #1281 is complete.

**Comment by ivan-mogilko** [comment]
> I replied to the greater problem in #1281 including the thumbnail idea. This issue here can probably be closed now

Thumbnails and other perfomance improvements in the editor are tasks separate from the sprite storage method. This problem is related to the editor, while sprite storage is a part of general workflow, which is related to both editor, and game compiling tools we are making in parallel.
It's best to keep this open, only keeping in mind that it's likely be addressed after other bigger changes are made.
